plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -74,19 +92,22 @@ public DefaultThriftMetastoreClientFactory(
             Duration connectTimeout,
             Duration readTimeout,
             HiveMetastoreAuthentication metastoreAuthentication,
-            String hostname)
+            String hostname,
+            Optional<ThriftHttpContext> thriftHttpContext)
     {
         this.sslContext = requireNonNull(sslContext, "sslContext is null");
         this.socksProxy = requireNonNull(socksProxy, "socksProxy is null");
         this.connectTimeoutMillis = toIntExact(connectTimeout.toMillis());
         this.readTimeoutMillis = toIntExact(readTimeout.toMillis());
         this.metastoreAuthentication = requireNonNull(metastoreAuthentication, "metastoreAuthentication is null");
         this.hostname = requireNonNull(hostname, "hostname is null");
+        this.thriftHttpContext = requireNonNull(thriftHttpContext, "thriftHttpContext is null");
     }
 
     @Inject
     public DefaultThriftMetastoreClientFactory(
             ThriftMetastoreConfig config,
+            ThriftHttpMetastoreConfig httpMetastoreConfig,
```

Comment:
This should be `Optional<ThriftHttpMetastoreConfig>` - non-empty in case of http scheme and empty in case of thrift scheme.

Reply:
Changed to Optional<ThriftHttpMetastoreConfig>